src/components/dialog/dialog.js

```diff
@@ -658,7 +659,9 @@ function MdDialogProvider($$interimElementProvider) {
         $mdDialog.cancel();
       };
       this.keypress = function($event) {
-        if ($event.keyCode === $mdConstant.KEY_CODE.ENTER) {
+        var invalidPrompt = isPrompt && this.required && !this.result;
```

Comment:
I don't think that this check is accurate enough. The input could be asking for the number of children and 0 would be a valid response. However `!this.result` would be false and flag the result as invalid in that case. `angular.isDefined(this.result)` is a safer approach and is used in many other components.

Reply:
agreed